rj_gameplay/rj_gameplay/gameplay_node.py

```diff
@@ -213,15 +216,22 @@ def update_world_state(self) -> None:
             assert self.world_state is not None
 
     def gameplay_tick(self) -> None:
+        """
+        Get situation, play from self.play_selector and update the currently running play if needed.
+        Then, add field and game_info to world_state, and push global obstacles to motion planning.
+        """
         self.update_world_state()
 
         if self.world_state is not None:
-            if self.test_play is None:
-                curr_situation, curr_play = self.play_selector.select(self.world_state)
-                intents = curr_play.tick(self.world_state)
-            else:
-                intents = self.test_play.tick(self.world_state)
-                curr_play = self.test_play
+            new_situation, new_play = self.play_selector.select(self.world_state)
+
+            if type(self._curr_play) != type(new_play) or type(
```

Comment:
nitpick: typically its safer to use 'is not' rather != when dealing with types, so it should be
```type(self._curr_play) is not type(new_play)```